Pear/Controller/ChatViewController.swift

```diff
@@ -0,0 +1,331 @@
+    }
+
+    @objc func backPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    private func hideKeyboardWhenViewTapped() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        tap.cancelsTouchesInView = false
+        tap.delegate = self
+        view.addGestureRecognizer(tap)
+    }
+
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    private func setupChatInput() {
+        chatInputContainerView.backgroundColor = .backgroundLightGreen
+        view.addSubview(chatInputContainerView)
+
+        chatInputTextField.attributedPlaceholder = NSAttributedString(string: "Say hi", attributes: [NSAttributedString.Key.foregroundColor: UIColor.greenGray, NSAttributedString.Key.font: UIFont.getFont(.book, size: 16)])
+        chatInputTextField.layer.cornerRadius = 12
+        chatInputTextField.backgroundColor = .white
+        chatInputTextField.delegate = self
+        chatInputContainerView.addSubview(chatInputTextField)
+
+        let leftPaddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 0))
+        chatInputTextField.leftView = leftPaddingView
+        chatInputTextField.leftViewMode = .always
+
+        let rightPaddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 0))
+        chatInputTextField.rightView = rightPaddingView
+        chatInputTextField.rightViewMode = .always
+
+        sendMessageButton.setImage(UIImage(named: "sendMessage"), for: .normal)
+        sendMessageButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
+        chatInputContainerView.addSubview(sendMessageButton)
+
+        separatorView.backgroundColor = .inactiveGreen
+        chatInputContainerView.addSubview(separatorView)
+    }
+
+    private func getChatMessages() {
+        databaseRef.child("User-Messages").child(currentUser.netID).child(messageUser.netID).observe(.childAdded) { snapshot in
+            let messageId = snapshot.key
+            let messageRef = self.databaseRef.child("Messages").child(messageId)
+            messageRef.observeSingleEvent(of: .value) { snapshot in
+                self.getMessages(snapshot: snapshot)
+            }
+        }
+    }
+
+    private func getMessages(snapshot: DataSnapshot) {
+        let message = PearMessage(snapshot: snapshot)
+        self.messages.append(message)
+        DispatchQueue.main.async {
+            self.groupMessagesByDate()
+        }
+    }
+
+    private func groupMessagesByDate() {
+        self.groupedMessagesByDate = []
+        let groupedMessages = Dictionary(grouping: messages) { message -> Date in
+            var day = NSDate()
+            if let timeSince1970 = Double(message.time) {
+                day = NSDate(timeIntervalSince1970: timeSince1970)
+            }
+            var date = day as Date
+            date = stripTimeOffDate(from: date)
+            return date
```

Comment:
`stripTimeOffDate(from:) --> stripTimeOff(date:)`

```suggestion
            var date = Date()
            if let timeSince1970 = Double(message.time) {
                date = Date(timeIntervalSince1970: timeSince1970)
            }
            return stripTimeOffDate(from: date)
```